### DISCUSSION TITLE
Can the header updated to create a broken packet?

### DISCUSSION THREAD

**Discussion by matsduf** [coredns/coredns#6394] [Q&A]
We have seen, out in the wild, DNS packets that are marked as truncated (TC flag set) and with authority counter greater than 0, but with no records available. For testing Zonemaster, a domain name and DNS testing tool (https://github.com/zonemaster/zonemaster) we want to be able to create such packets with the help of e.g. CoreDNS.

The Header plugin lets you set some flags, but not the record counters. Is such a plugin available?

**Comment by chrisohaver** [comment]
CoreDNS does not include a plugin that can do this. As far as developing a plugin that does, I'm not sure if the miekg/dns package allows such level of control.  I suspect the package calculates the record counters by actually counting the records in the payload, and that they are not directly settable.
